File: c#/src/Sodium/Sodium.Tests.Internal/NodeTests.cs
using NUnit.Framework;

namespace Sodium.Tests.Internal
{
    [TestFixture]
    public class NodeTests
    {
        [Test]
        public void TestNode()
        {
            Node<int> a = new Node<int>();
            Node<int> b = new Node<int>();
            Transaction.Apply(trans =>
            {
                a.Link(trans, (t, v) => { }, b);
                trans.Prioritized(a, t => { });
                return Unit.Value;
            }, false);
            Assert.That(a.Rank, Is.LessThan(b.Rank));
        }
    }
}

File: f#/src/Sodium/Sodium.Tests.Internal/NodeTests.fs
module Sodium.Tests.Internal.Node

open NUnit.Framework
open Sodium

[<TestFixture>]
type Tests() =

    [<Test>]
    member __.``Test Node``() =
        let a = Node<int>(0L)
        let b = Node<int>(1L)
        a.Link (fun t v -> ()) b |> ignore
        Assert.That(a, Is.LessThan(b))
